source/MissionAction.cpp

```diff
@@ -419,9 +426,14 @@ void MissionAction::Do(PlayerInfo &player, UI *ui, const System *destination, co
 			subs["<ship>"] = player.Flagship()->Name();
 		string text = Format::Replace(dialogText, subs);
 		
+		// Don't push the dialog text if this is a visit action on a nonunique
+		// mission; on visit, nonunique dialogs are handled by PlayerInfo as to
+		// avoid the player being spammed by dialogs if they have multiple
+		// missions active with the same destination (e.g. in the case of
+		// stacking bounty jobs).
 		if(isOffer)
 			ui->Push(new Dialog(text, player, destination));
-		else
+		else if(!(trigger == "visit" && !isUnique))
```

Comment:
Is `else if(isUnique || trigger != "visit")` a clearer way to state the condition?

Reply:
Idk, would it be to you? I was thinking of simplifying the Boolean algebras here but wondered if this way was easier to read.

Reply:
The double negative makes the statement relatively hard to read; Tehhowchs version is easier to read for somebody seeing the code for the first time.